Register diag_backward, diagonal_backward, infinitetely...gelu_backward as operators

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #44505 Batched grad support for: slice, select, diagonal
* #44494 Add some batched gradient tests
* **#44422 Register diag_backward, diagonal_backward, infinitetely...gelu_backward as operators**
* #44410 Register cummaxmin_backward, cumprod_backward as operators
* #44408 Add trace_backward, masked_select_backward, and take_backward as ops

See #44052 for context.

Test Plan:
- `pytest test/test_autograd.py -v`
- `pytest test/test_nn.py -v`

Differential Revision: [D23607691](https://our.internmc.facebook.com/intern/diff/D23607691)

zou3519 committed:
Register diag_backward, diagonal_backward, infinitetely...gelu_backward as operators

See #44052 for context.

Test Plan:
- `pytest test/test_autograd.py -v`
- `pytest test/test_nn.py -v`

[ghstack-poisoned]

albanD commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -2738,6 +2743,12 @@
+- func: infinitely_differentiable_gelu_backward(Tensor grad, Tensor self) -> Tensor
+  use_c10_dispatcher: full
+  variants: function
+  python_module: nn
```
Is that important? This function shout not be exposed to python anyways right?

zou3519 replied:
Yup, this doesn't do anything. I put it in to make it consistent with all of the other registered nn backward functions (those have a python_module: nn that doesn't do anything). E.g., 
https://github.com/pytorch/pytorch/blob/367426494759ddde0896665ed55c6f9af2870cf0/aten/src/ATen/native/native_functions.yaml#L2726
https://github.com/pytorch/pytorch/blob/367426494759ddde0896665ed55c6f9af2870cf0/aten/src/ATen/native/native_functions.yaml#L701

albanD reviewed (approved):
LGTM

zou3519 committed:
Update on "Register diag_backward, diagonal_backward, infinitetely...gelu_backward as operators"


See #44052 for context.

Test Plan:
- `pytest test/test_autograd.py -v`
- `pytest test/test_nn.py -v`

Differential Revision: [D23607691](https://our.internmc.facebook.com/intern/diff/D23607691)

[ghstack-poisoned]

zou3519 committed:
Update on "Register diag_backward, diagonal_backward, infinitetely...gelu_backward as operators"


See #44052 for context.

Test Plan:
- `pytest test/test_autograd.py -v`
- `pytest test/test_nn.py -v`

Differential Revision: [D23607691](https://our.internmc.facebook.com/intern/diff/D23607691)

[ghstack-poisoned]
